test/atomic_bitset.cpp

```diff
@@ -0,0 +1,377 @@
+#include <stdx/atomic_bitset.hpp>
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include <cstdint>
+#include <thread>
+#include <type_traits>
+
+TEST_CASE("atomic_bitset with explicit storage element type",
+          "[atomic_bitset]") {
+    static_assert(sizeof(stdx::atomic_bitset<1, std::uint8_t>) ==
+                  sizeof(std::uint8_t));
+    static_assert(sizeof(stdx::atomic_bitset<8, std::uint8_t>) ==
+                  sizeof(std::uint8_t));
+    static_assert(sizeof(stdx::atomic_bitset<1, std::uint16_t>) ==
+                  sizeof(std::uint16_t));
+    static_assert(sizeof(stdx::atomic_bitset<16, std::uint16_t>) ==
+                  sizeof(std::uint16_t));
+    static_assert(sizeof(stdx::atomic_bitset<1, std::uint32_t>) ==
+                  sizeof(std::uint32_t));
+    static_assert(sizeof(stdx::atomic_bitset<32, std::uint32_t>) ==
+                  sizeof(std::uint32_t));
+    static_assert(sizeof(stdx::atomic_bitset<1, std::uint64_t>) ==
+                  sizeof(std::uint64_t));
+    static_assert(sizeof(stdx::atomic_bitset<64, std::uint64_t>) ==
+                  sizeof(std::uint64_t));
+}
+
+TEST_CASE("atomic_bitset with implicit storage element type",
+          "[atomic_bitset]") {
+    static_assert(sizeof(stdx::atomic_bitset<1>) == sizeof(std::uint8_t));
+    static_assert(sizeof(stdx::atomic_bitset<8>) == sizeof(std::uint8_t));
+    static_assert(sizeof(stdx::atomic_bitset<9>) == sizeof(std::uint16_t));
+    static_assert(sizeof(stdx::atomic_bitset<16>) == sizeof(std::uint16_t));
+    static_assert(sizeof(stdx::atomic_bitset<17>) == sizeof(std::uint32_t));
+    static_assert(sizeof(stdx::atomic_bitset<32>) == sizeof(std::uint32_t));
+    static_assert(sizeof(stdx::atomic_bitset<33>) == sizeof(std::uint64_t));
+    static_assert(sizeof(stdx::atomic_bitset<64>) == sizeof(std::uint64_t));
+}
+
+#ifdef _LIBCPP_VERSION
+TEMPLATE_TEST_CASE("atomic_bitset is lock free", "[atomic_bitset]",
+                   std::uint8_t, std::uint16_t, std::uint32_t, std::uint64_t) {
+    CHECK(stdx::atomic_bitset<8, TestType>{}.is_lock_free());
+}
+#endif
```

Comment:
question: what's the check on the _LIBCPP_VERSION macro for?

Reply:
It's because libstdc++ doesn't link (!) when I use `is_lock_free`! But I think it's easiest to remove this since we really want `is_always_lock_free`.

Reply:
Removed.

Reply:
that's wild!